File: corpus/go-efficiency-design-v2/changed/enrichment.go
package enrich

type Owner struct { Enabled bool }
type Item struct { Active bool; Owner *Owner }
type Repository interface {
    Load(int64) Item
    LoadMany([]int64) []Item
}

type ResultSliceFactory struct{}
func (ResultSliceFactory) Create() []Item { return []Item{} }

func Enrich(ids []int64, repo Repository) []Item {
    results := (ResultSliceFactory{}).Create()
    for _, id := range ids {
        item := repo.Load(id)
        if item.Active {
            if item.Owner != nil {
                if item.Owner.Enabled { results = append(results, item) }
            }
        }
    }
    return results
}

File: corpus/typescript-efficiency-design-v2/base/enrichment.ts
interface Item { active: boolean; owner?: { enabled: boolean } }
interface Repository {
  load(id: string): Promise<Item>;
  loadMany(ids: string[]): Promise<Item[]>;
}

export async function enrich(ids: string[], repo: Repository): Promise<Item[]> {
  const results = [];
  for (const item of await repo.loadMany(ids)) {
    if (item.active && item.owner?.enabled) results.push(item);
  }
  return results;
}
